### PR TITLE
Overhaul javascript organisation, switch to typescript

### PR DESCRIPTION
**Pull request by mhauru**
This started as an effort to convert all our javascript to typescript. This required restructuring the code a lot, so in the process I also took care of many other long-standing javascript-related issues:
* Modernised how we use the datatables.net library
* Got rid of unnecessary dependencies
* Renamed many of the javascript files to be more helpful
* Oblitared the `custom.js` file, which included hundreds and hundreds of lines of dead copypasta code from CROP
* Made a bazillion small changes all over, both because they seemed helpful to me, and because the typescript compiler highlighted issues (hurray for type checking catching bugs!)

I appreciate that this is tricky to review since there's so much going on, happy to talk it through. One good bit to read is a new `README.md` in `webapp/base/static`, which explains the new organisation of what sort of code should go where.

EDIT: It seems that, very unhelpfully, GitHub does not realise that most of the typescript files are the old javascript files renamed and with small changes, which certainly doesn't make this any easier to review...

### CODE REVIEW

**Comment by mhauru** [comment]
The two failing tests will be fixed by this one: https://github.com/alan-turing-institute/DTBase/pull/180

**Comment by EdwinB12** [comment]
@GiorgioCerro - Think you might be more suited to reviewing this PR. I'll try and look over it tomorrow but think it would benefit from your review. 

@mhauru - I approved and merged #180 , could you merge it into this PR please so the tests pass?

**Comment by GiorgioCerro** [comment]
Great work Markus. I will have a deeper look at that either today or tomorrow!

File: dtbase/webapp/app/base/static/README.md
Comment (GiorgioCerro) [inline dtbase/webapp/app/base/static/README.md:1]: Excellent. I could go through all of it easily and it explains clearly  the use of .ts and .js files. 
I was just a bit confused on why we still need .js files, but the second paragraph clarifies it. 
Curiosity: does webpack create temporary files while the webapp is running or permanently files?

File: dtbase/webapp/run.sh
Comment (GiorgioCerro) [inline dtbase/webapp/run.sh:1]: No comments here. Everything's clear.

**Comment by GiorgioCerro** [comment]
Issues:
- when I run the webapp with DT_CONFIG_MODE="Auto-login" I get 'permission denied' and it doesnt open the page. Without I still get 'permission denied' but I can open the page
- I don't know if it's related, but when I go to the webpage and I create new sensors., I don't see them in the table list (it's empty).

File: dtbase/webapp/app/base/static/README.md
Comment (mhauru) [reply dtbase/webapp/app/base/static/README.md:1]: Permanent, they'll stay there. I've added them to .gitignore, but I do still find them a bit annoying.

**Review by GiorgioCerro — APPROVED** [review]
Awesome! I checked all the issues I encountered and it was my fault because I needed to add permission to execute the .sh file. 
Also, the table of sensors looks way better (now we can download the data and order them in any way we like)

**Pull request merged**